src/lib/stepper/_stepper-theme.scss

```diff
@@ -20,18 +21,38 @@
     }
 
     .mat-step-icon {
+      color: mat-color($primary, default-contrast);
+    }
+
+    .mat-step-icon-selected,
+    .mat-step-icon-state-done,
+    .mat-step-icon-state-edit {
+      background-color: mat-color($primary) !important;
+      color: mat-color($primary, default-contrast);
+    }
+
+    .mat-step-icon-state-number {
       background-color: mat-color($primary);
       color: mat-color($primary, default-contrast);
     }
 
-    .mat-step-icon-not-touched {
+    .mat-step-icon:not(.mat-step-icon-selected) {
       background-color: mat-color($foreground, disabled-text);
       color: mat-color($primary, default-contrast);
     }
 
+    .mat-step-icon-state-error {
+      background-color: transparent !important;
```

Comment:
We can't have `!important` anywhere in our CSS. I assume this is because you need greater specificity due to the `.mat-step-icon:not(.mat-step-icon-selected)` selector? Maybe instead we can just order them so that the right one takes precedence:

```scss
// default that will apply to every icon
.mat-step-icon {
  background-color: mat-color($foreground, disabled-text);
  color: mat-color($primary, default-contrast);
}

// override background for icons that are selected or are in the edit/done state
.mat-step-icon-selected,
.mat-step-icon-state-done,
.mat-step-icon-state-edit {
  background-color: mat-color($primary);
}

// override background and foreground for error state (will also take precedence over selected style)
.mat-step-icon-state-error {
  background-color: transparent;
  color: mat-color($warn);
}
```

I _think_ the above is the correct order, but you should confirm. See https://www.w3schools.com/css/css_specificity.asp for more about CSS selector specificity

Reply:
Ok I was able to remove the `!important` by using css specificity but it would not work with the `not()` selector so I had to use the `-not-selected` class again.

Reply:
Ok, I pushed code that I think should be equivalent to what you had but doesn't rely on the class

Reply:
Perfect!